src/main/java/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -1,21 +1,24 @@
 package cinema.security;
 
 import cinema.exception.AuthenticationException;
-import cinema.lib.Inject;
-import cinema.lib.Service;
 import cinema.model.User;
 import cinema.service.ShoppingCartService;
 import cinema.service.UserService;
 import cinema.util.HashUtil;
 import java.util.Optional;
+import org.springframework.stereotype.Service;
 
 @Service
 public class AuthenticationServiceImpl implements AuthenticationService {
-    @Inject
-    private UserService userService;
-    @Inject
+    private final UserService userService;
     private ShoppingCartService shoppingCartService;
```

Comment:
why do we use `final ` only with one field in this class?

Reply:
Fixed